src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs

```diff
@@ -1426,5 +1713,131 @@ .maxstack  1
   IL_001b:  ret
 }");
         }
+
+        [Fact]
+        public void DataPropertiesInitOnly()
+        {
+            var src = @"
+class C
+{
+    data int X;
+
+    public static void Main()
+    {
+        var c = new C() { X = 0 };
+        c.X = 2;
+    }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (9,9): error CS8852: Init-only property or indexer 'C.X' can only be assigned in an object initializer, or on 'this' or 'base' in an instance constructor or an 'init' accessor.
+                //         c.X = 2;
+                Diagnostic(ErrorCode.ERR_AssignmentInitOnly, "c.X").WithArguments("C.X").WithLocation(9, 9));
+        }
+
+        [Fact]
+        public void DataPropertiesEmit()
+        {
+            var src = @"
+using System;
+class C
+{
+    data int X;
+
+    public static void Main()
+    {
+        var c = new C() { X = 5 };
+        Console.WriteLine(c.X);
+    }
+}";
+            var verifier = CompileAndVerify(src, expectedOutput: "5");
+            verifier.VerifyIL("C.Main", @"
+{
+  // Code size       23 (0x17)
+  .maxstack  3
+  IL_0000:  newobj     ""C..ctor()""
+  IL_0005:  dup
+  IL_0006:  ldc.i4.5
+  IL_0007:  callvirt   ""void C.X.init""
+  IL_000c:  callvirt   ""int C.X.get""
+  IL_0011:  call       ""void System.Console.WriteLine(int)""
+  IL_0016:  ret
+}");
```

Comment:
Sorry, I should have been clearer. I meant verify the IL of the accessors (since this is the new code we're emitting).